sotodlib/hwp/hwp.py

```diff
@@ -911,3 +911,121 @@ def get_tau_hwp(
+
+    At the boundary of subscan_1 and subscan_2 (with constant coefficients c_s1, c_s2, respectively):
+    c(t)_i = {c_s1 for s < s_pivot - n_buf
+              c_s2 + (c_s1 - c_s2)*wn(s) for  s_pivot - n_buf <= s <= s_pivot + n_buf
+              c_s2 for s > s_pivot + n_buf}
+
+    Here, wn(s) is the window function chosen to span 2*n_buf for a smooth transition between coefficients, 
+    and s_pivot is the single sample transition point between subscan_1 and subscan_2.
+    '''
+    if f'{hwpss_model_name}' in aman._assignments:
+        if not replace_model:
+            raise ValueError('hwpss_model_name must be different than what is already in aman if replace_model=False')
+        else:
+            aman.move(hwpss_model_name,None)
+    
+    aman.wrap(hwpss_model_name, np.zeros_like(aman.signal,dtype='float32'),[(0, 'dets'), (1, 'samps')])
+
+    #re-calculate subscan flags with turnarounds included, for apodization purposes
+    subscan_info = tod_ops.flags.get_subscans(aman, flags=None, merge=False, include_turnarounds=True, overwrite=False)
+    num_subscans = len(subscan_info.subscan_flags)
+
+    #transition_flag = np.zeros_like(aman.hwpss_model)
+    coeffs = np.zeros(shape=(aman.dets.count,2*len(modes),len(subscan_info.subscan_flags))) #list of hwpss_subscan harmonic coeff
+    transition_indices = np.array([])
+
+    for j in range(num_subscans):
+        subscan_flag = subscan_info.subscan_flags[j]
+        i = subscan_flag.ranges()[0]
+        sub_aman = aman.restrict('samps', (i[0] + aman.samps.offset, i[1] + aman.samps.offset), in_place=False)
+        sub_aman.move(hwpss_model_name, None)
+        if flags is None:
+            sub_flag = RangesMatrix.from_mask(np.zeros(aman.samps.count, dtype=int)[i[0]:i[1]])
+        else:
+            sub_flag = RangesMatrix.from_mask(flags.mask()[:, i[0]:i[1]])
+
+        
+        get_hwpss(
+            aman=sub_aman,
+            flags=sub_flag,
+            signal=signal,
+            hwp_angle=hwp_angle,
+            bin_signal=bin_signal,
+            bins=bins,
+            lin_reg=lin_reg,
+            modes=modes,
+            apply_prefilt=apply_prefilt,
+            prefilt_cfg=prefilt_cfg,
+            prefilt_detrend=prefilt_detrend,
+            apodize_edges=apodize_edges,
+            apodize_edges_samps=apodize_edges,
+            apodize_flags=apodize_flags,
+            apodize_flags_samps=apodize_flags,
+            merge_stats=True,
+            merge_model=True,
+            hwpss_model_name=hwpss_model_name,
+        )
+        aman.get(hwpss_model_name)[:, i[0]:i[1]] = sub_aman.hwpss_model
+        coeffs[:,:,j]+=sub_aman.hwpss_stats.coeffs
+        del sub_aman
+
+        transition_msk = aman.flags.turnarounds.mask() & subscan_flag.mask()
+        transition_idx = np.where(np.diff(transition_msk.astype(int)) == -1)[0] #determining index of falling edge which denotes transition to the next subscan 
```

Comment:
You should be able to do this as RangesMatrices as well. Add in Ranges and then you will get a range which is the samples you care about and don't have to look for diff = -1.